### PR TITLE
[MBL-16882][Student] HTML Parsing

### PR DESCRIPTION
**Pull request by tamaskozmer**
Test plan:
- Verify that all the images are displayed correctly in HTML content (pages, assignment details, quiz details, syllabus, syllabus events)
- Verify that the above behavior is correct also when file sync is not checked.
- Check that the progress is correct on the offline sync progress screen.
- Verify that all the external and internal files are cleaned up when sync is disabled for a course.
- Discussions and announcements should work only after the discussion/announcement details ticket.

refs: MBL-16682
affects: Student
release note: none

## Checklist

- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kristofnemere — APPROVED** [review]
QA+1

File: libs/pandautils/src/main/java/com/instructure/pandautils/di/ApplicationModule.kt [hunk 1]
```diff
@@ -98,4 +104,16 @@ class ApplicationModule {
     fun provideStorageUtils(@ApplicationContext context: Context): StorageUtils {
         return StorageUtils(context)
     }
+
+    @Provides
+    fun provideHtmlParses(
```
Comment (hermannakos) [inline libs/pandautils/src/main/java/com/instructure/pandautils/di/ApplicationModule.kt:109]: Typo

Comment (hermannakos) [inline libs/pandautils/src/main/java/com/instructure/pandautils/di/ApplicationModule.kt:109]: Typo

**Review by hermannakos — APPROVED** [review]
QA + 1

**Pull request merged**